olp-cpp-sdk-dataservice-read/src/VersionedLayerClientImpl.cpp

```diff
@@ -320,6 +327,40 @@ VersionedLayerClientImpl::PrefetchTiles(PrefetchTilesRequest request) {
                                                           promise);
 }
 
+VersionedLayerClientImpl::VersionResponse VersionedLayerClientImpl::GetVersion(
+    boost::optional<int64_t> requested_version,
```

Comment:
Not quite, but close :)
The main idea is that the version is locked either by the user in the constructor or by the first call to GetLatestVersion() triggered by either one of the API calls from VersionedLayerClient.
This means that once the user can specify the version or boost::none in the constructor we will deprecate all the WithVersion() from the request classes and we should ignore them.